File: leet_code/cpp/58.最后一个单词的长度.cpp
/*
 * @lc app=leetcode.cn id=58 lang=cpp
 *
 * [58] 最后一个单词的长度
 */
#include "common.h"
// @lc code=start
class Solution {
public:
  int lengthOfLastWord(string s) {
    if (s.empty()) {
      return 0;
    }
    int i = s.size() - 1;
    while (s[i] == ' ' && i > 0) {
      i--;
    }
    if (i < 0)
      return 0;
    int j = i;
    for (; j >= 0 && s[j] != ' '; --j) {
    }
    return i - j;
  }
};
// @lc code=end

File: python/58.最后一个单词的长度.py
#
# @lc app=leetcode id=58 lang=python3
#
# [58] Length of Last Word
#
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        ls = len(s)
        # slow and fast pointers
        slow = -1
        # iterate over trailing spaces
        while slow >= -ls and s[slow] == ' ':
            slow-=1
        fast = slow
        # iterate over last word
        while fast >= -ls and s[fast] != ' ':
            fast-=1
        return slow - fast

